tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetryCollector.cs

```diff
@@ -0,0 +1,175 @@
+﻿// <copyright file="ConfigurationTelemetryCollector.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Reflection;
+using System.Threading;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.PlatformHelpers;
+
+namespace Datadog.Trace.Telemetry
+{
+    internal class ConfigurationTelemetryCollector
+    {
+        private int _tracerInstanceCount = 0;
+        private int _hasChangesFlag = 0;
+        private volatile CurrentSettings _settings;
+        private volatile SecuritySettings _securitySettings;
+        private volatile bool _isTracerInitialized = false;
+        private AzureAppServices _azureApServicesMetadata;
+        private HostTelemetryData _hostData = null;
+
+        public void RecordTracerSettings(
+            ImmutableTracerSettings tracerSettings,
+            string defaultServiceName,
+            AzureAppServices appServicesMetadata)
+        {
+            // Increment number of times this has been called
+            var reconfigureCount = Interlocked.Increment(ref _tracerInstanceCount);
+            var appData = new ApplicationTelemetryData
+            {
+                ServiceName = defaultServiceName,
+                Env = tracerSettings.Environment,
+                ServiceVersion = tracerSettings.ServiceVersion,
+                TracerVersion = TracerConstants.AssemblyVersion,
+                LanguageName = "dotnet",
+                LanguageVersion = FrameworkDescription.Instance.ProductVersion,
+                RuntimeName = FrameworkDescription.Instance.Name,
+            };
+
+            _settings = new CurrentSettings(tracerSettings, appData);
+
+            // The remaining properties can't change, so only need to set them the first time
+            if (reconfigureCount != 1)
```

Comment:
It looks like this method can be invoked multiple times. Should the comparison here be `reconfigureCount >= 1`?